src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
> The 411 is what we want, but how do we communicate that 411 back to the code without throwing an exception, or passing in the Reponse instance

We don't need to. reactphp/http is abstraction for the http protocol, which means it should be able to return own responses when a client request is made that is either not spec conform or when the http server doesn't understand fundamental protocol encoding like the current topic here. As a user of reactphp/http I have no benefit in knowing that a client made a bad request. My application should only get valid requests. To get a valid request reactphp/http should do whatever is necessary with the client based on http specs, so he sends us valid stuff. (If sending a 411/500/xxx response is part of this, then so be it)